fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerConflictsTest.java

```diff
@@ -201,5 +201,60 @@ public void run() {
         assertSame(recordStore.getIndexState(index.getName()), IndexState.READABLE);
         assertEquals(0, timer.getCount(FDBStoreTimer.Events.COMMIT_FAILURE));
     }
+
+    @Test
+    public void testOnlineIndexerIdempotentNested() throws Exception {
+        // If a conflict happens, this test should succeed by retries
+        Random rand = new Random();
+        final long seed = rand.nextInt();
+        LOGGER.info("Random seed is " + seed);
+        rand = new Random(seed); // make rand repeatable (note that multi threading behaviour is not fully repeatable in this test)
+
+        List<TestRecords1Proto.MySimpleRecord> records =
+                LongStream.range(0, 400).mapToObj(val -> TestRecords1Proto.MySimpleRecord.newBuilder().setRecNo(val).setNumValue2((int)val + 1).build()
+                ).collect(Collectors.toList());
+
+        Index index = new Index("newIndex", field("num_value_2"), IndexTypes.VALUE);
+        FDBRecordStoreTestBase.RecordMetaDataHook hook = metaDataBuilder -> metaDataBuilder.addIndex("MySimpleRecord", index);
+        openSimpleMetaData();
+        try (FDBRecordContext context = openContext()) {
+            for (int i = 50; i <= 150; i++) {
+                // even numbers 100-300 (that's 101 values)
+                recordStore.saveRecord(records.get(i * 2));
+            }
+            context.commit();
+        }
+
+        openSimpleMetaData(hook);
+        try (FDBRecordContext context = openContext()) {
+            recordStore.markIndexWriteOnly(index.getName()).join();
+            context.commit();
+        }
+
+        int[] inserts = {99, 101, 133, 201, 350, 299, 301};
+        for (int i = 0; i < inserts.length; i++) {
+            int record_i = inserts[i];
+
+            final FDBStoreTimer timer = new FDBStoreTimer();
+            LOGGER.info("newIndex thread run");
+            try (FDBRecordContext context1 = openContext()) {
+                try (OnlineIndexer indexer =
+                             OnlineIndexer.newBuilder()
+                                     .setRecordStore(recordStore)
+                                     .setIndex("newIndex")
+                                     .setTimer(timer)
+                                     .build()) {
+                    indexer.buildRange(recordStore, null, null).join();
+                    try (FDBRecordContext context2 = openContext()) {
+                        recordStore.saveRecord(records.get(record_i));
+                        context2.commit();
+                    }
+                    context1.commit();
+                }
+            }
+
+            assertEquals(0, timer.getCount(FDBStoreTimer.Events.COMMIT_FAILURE));
```

Comment:
This assert isn't hurting anything, I suppose, but it's also already covered by the two contexts committing successfully above. It may be worth adding a comment on `context1.commit()` about how a record is being added in the range added by the indexer, but the transaction still commits because it's value isn't needed by the indexer.